tests/garage/torch/policies/test_gaussian_mlp_policy.py

```diff
@@ -0,0 +1,105 @@
+import pickle
+from unittest import mock
+
+import numpy as np
+import pytest
+import torch
+
+from garage.torch.modules import GaussianMLPModule, GaussianMLPSharedModule, GaussianMLPAdaptiveModule
+from garage.torch.policies import GaussianMLPPolicy
+
+
+test_settings = [
+    (1, 1, (1,)),
+    (1, 2, (2,)),
+    (1, 3, (3,)),
+    (1, 1, (1, 2)),
+    (1, 2, (2, 1)),
+    (1, 3, (4, 5)),
+    (2, 1, (1,)),
+    (2, 2, (2,)),
+    (2, 3, (3,)),
+    (2, 1, (1, 2)),
+    (2, 2, (2, 1)),
+    (2, 3, (4, 5)),
+    (5, 1, (1,)),
+    (5, 2, (2,)),
+    (5, 3, (3,)),
+    (5, 1, (1, 2)),
+    (5, 2, (2, 1)),
+    (5, 3, (4, 5)),
+]
+
+
+@pytest.mark.parametrize("input_dim, output_dim, hidden_sizes", test_settings)
+@mock.patch('garage.torch.modules.GaussianMLPModule')
+def test_policy_get_actions(mock_model, input_dim, output_dim, hidden_sizes):
+    action = torch.randn((output_dim,))
+
+    mock_dist = mock.MagicMock()
+    mock_dist.rsample.return_value = action
+
+    mock_model.return_value = mock_dist
+
+    env_spec = mock.MagicMock()
+    env_spec.observation_space.flat_dim = input_dim
+    env_spec.action_space.flat_dim = output_dim
+
+    policy = GaussianMLPPolicy(env_spec, mock_model)
+
+    input = torch.ones(input_dim)
+    sample = policy.get_actions(input)
+
+    assert np.array_equal(sample, action.detach().numpy())
+
+
+@pytest.mark.parametrize("input_dim, output_dim, hidden_sizes", test_settings)
+@mock.patch('garage.torch.modules.GaussianMLPModule')
+def test_policy_get_action(mock_model, input_dim, output_dim, hidden_sizes):
+    action = torch.randn((output_dim,))
+
+    mock_dist = mock.MagicMock()
+    mock_dist.rsample.return_value = action
+
+    mock_model.return_value = mock_dist
+    input_dim, output_dim, hidden_sizes = (5, 3, (2, 2))
+
+    env_spec = mock.MagicMock()
+    env_spec.observation_space.flat_dim = input_dim
+    env_spec.action_space.flat_dim = output_dim
+
+    policy = GaussianMLPPolicy(env_spec, mock_model)
+
+    input = torch.ones(input_dim)
+    sample = policy.get_action(input)
+
+    assert sample == action[0].item()
+
+
+@pytest.mark.parametrize("input_dim, output_dim, hidden_sizes", test_settings)
+def test_policy_is_picklable(input_dim, output_dim, hidden_sizes):
+    env = EnvForTest(SpaceForTest(input_dim), SpaceForTest(output_dim))
+
+    module = GaussianMLPModule(input_dim, output_dim, hidden_sizes=hidden_sizes)
+    policy = GaussianMLPPolicy(env, module)
+
+    input = torch.ones(input_dim)
+    sample = policy.get_actions(input)
+
+    h = pickle.dumps(policy)
+    policy_pickled = pickle.loads(h)
+
+    sample_p = policy_pickled.get_actions(input)
+
+    assert np.array_equal(sample.shape, sample_p.shape)
+
+
+class EnvForTest:
```

Comment:
we have existing fixtures for this in tests/fixtures. I also think mocks would probably work fine.

Reply:
delete this code because the pickable testcase is removed